addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/SpringActuatorScanRule.java

```diff
@@ -0,0 +1,193 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import org.apache.commons.httpclient.URI;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractHostPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+/**
+ * An active scan rule to test for Spring Actuators revealing too much info. Health might be the
+ * most commonly enabled but if health is enabled, so might be Thread Dump or Heap Dump which could
+ * be serious https://docs.spring.io/spring-boot/docs/current/actuator-api/htmlsingle/#overview
+ *
+ * @author sgerlach
+ */
+public class SpringActuatorScanRule extends AbstractHostPlugin {
+
+    private static final String MESSAGE_PREFIX = "ascanalpha.springactuator.";
+    private static final int PLUGIN_ID = 40042;
+    private static final Logger LOG = LogManager.getLogger(SpringActuatorScanRule.class);
+    private static final Pattern CONTENT_TYPE =
+            Pattern.compile(
+                    "application\\/vnd\\.spring-boot\\.actuator\\.v[0-9]\\+json|application\\/json",
+                    Pattern.MULTILINE);
+    private static final Pattern JSON_PAYLOAD = Pattern.compile("\\{.*\\:.*\\}", Pattern.MULTILINE);
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        // TODO change this to Spring, but need to extend the Tech in Core ZAP first
+        return technologies.includes(Tech.JAVA);
+    }
+
+    @Override
+    public int getId() {
+        return PLUGIN_ID;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.INFO_GATHER;
+    }
+
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+    }
+
+    @Override
+    public int getRisk() {
+        return Alert.RISK_MEDIUM;
+    }
+
+    @Override
+    public int getWascId() {
+        return 13;
+    }
```

Comment:
No cwe? Could be 215 : Insertion of Sensitive Information Into Debugging Code - some of the other similar rules use that.
You could now add the OWASP tags too, but def not a requirement ;)

Reply:
happy to ad the tags if you show me how to do it. Since this just got released I'd love to add it in later. I'll get the CWE in there.

Reply:
Heres one of my PRs that does that https://github.com/zaproxy/zap-extensions/pull/3180
Dont worry, you should only need 2 changes - the *.gradle.kts one and then implementing `getAlertTags()` - just ping me if you have any questions about that...

Reply:
I've done the release rules, I'll work through the beta then alpha ones when I get the chance, so its def not a priority unless you really want to see it in there ;)